pkcs7/pkcs7_chain_test.go

```diff
@@ -0,0 +1,203 @@
+package pkcs7
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"testing"
+	"time"
+)
+
+// Helper function to create a certificate chain for testing
+func createTestCertChain(t *testing.T) (*rsa.PrivateKey, *x509.Certificate, []*x509.Certificate) {
```

Comment:
This function is repeated twice.

I suggest we make a new package called `internal/certtest` and export those functions there. Then we can share them freely in the code without having to care about them being exported. See my `go-tpm-keyfiles` library for an example.

https://github.com/Foxboron/go-tpm-keyfiles/blob/master/internal/keytest/keytest.go